chapters/synchronization.txt

```diff
@@ -53,67 +53,133 @@ _memory dependencies_ between two sets of operations defined by the
+_happen-before_ the second dependency's *Ops~dst~*.
+This dependency can: itself also form a chain with another execution
+dependency, which can form an _execution dependency chain_ of more than two
+depenencies.
+_Execution dependency chain_ complements the specification of individual
+execution dependencies with the following description:
+
+  * Let *S~1~*, *S~2~*, ..., *S~N~* be a list of *N* synchronization
+    commands submitted in this order.
+  * Let's assume *S~1~*, *S~2~*, ..., *S~N~* are all part of an
+    _execution dependency chain_ *EC* such that *S~N~* is directly or
+    transitively chained with *S~1~*.
+  * Let *Scope~src~* be the first synchronization scope of *S~1~*.
+  * Let *Scope~dst~* be the second synchronization scope of *S~N~*.
+  * The chain *EC* introduces same
+    <<synchronization-dependencies-execution,execution dependency>>
+    guarantees as if instead an individual synchronization command *S* was
+    executed with these *Scope~src~* and *Scope~dst~* scopes.
 
 Execution dependencies alone are not sufficient to guarantee that values
 resulting from writes in one set of operations can: be read from another set
 of operations.
 
 [[synchronization-dependencies-available-and-visible]]
-Three additional types of operations are used to control memory access.
-_Availability operations_ cause the values generated by specified memory
-write accesses to become _available_ to a memory domain for future access.
-Any available value remains available until a subsequent write to the same
-memory location occurs (whether it is made available or not) or the memory
-is freed.
-_Memory domain operations_ cause writes that are available to a source
-memory domain to become available to a destination memory domain (an example
-of this is making writes available to the host domain available to the
-device domain).
-_Visibility operations_ cause values available to a memory domain to become
-_visible_ to specified memory accesses.
+Three additional types of operations are used to control memory access:
+
+  * _Availability operations_ cause the values generated by specified memory
+    write accesses to become _available_ in a specific memory domain for
+    future access.
+    Any available value remains available until a subsequent write to the
+    same memory location occurs (whether it is made available or not) or the
+    memory is freed.
+
+  * _Memory domain operations_ cause writes that are available in a source
+    memory domain to become available in a destination memory domain (an
+    example of this is making writes available in the host domain available
+    in the device domain).
+
+  * _Visibility operations_ cause values available in a specific memory
+    domain to become _visible_ to specified memory accesses.
+
+ifdef::editing-notes[]
+[NOTE]
+.editing-note
+====
+It feels the following is kinda important and should be part of core 1.0
+Vulkan to some extent.
+====
+endif::editing-notes[]
```

Comment:
At some point we hope to better integrate the memory model appendix with the synchronization chapter in general, it's just not something we've had the resource to do, _yet_ - I'm not sure an editors note here is necessary as this is tracked internally, but I'm not particularly against including it. For now we've just hidden it behind the extension as some of the definitions only _somewhat_ apply without it.

Mostly just adding some context here, not necessarily suggesting a change.

Reply:
The happens-before\after is frequently used, so should be integrated with the core spec sooner than later...